release-calendar/src/client/models/release-event.ts

```diff
@@ -38,11 +39,19 @@ export function getAllReleasesEvents(
 }
 
 export function getSingleReleaseEvents(
-  eventInputs: ReleaseEventInput[],
-): EventSourceInput[] {
-  return eventInputs.map(
-    (event: ReleaseEventInput): EventSourceInput => {
-      return {events: [event], textColor: 'white'};
-    },
-  );
+  SingleReleasePromotions: SingleReleasePromotion[],
+  allReleases: Map<Channel, ExtendedEventSourceInput>,
+): ExtendedEventSourceInput[] {
+  return SingleReleasePromotions.map((promotion: SingleReleasePromotion) => {
+    return {
+      events: [
+        (allReleases.get(promotion.channel).events as EventInput[]).find(
+          (event) => {
+            return event.start == promotion.date;
+          },
+        ),
+      ],
+      textColor: 'white',
+    };
+  });
```

Comment:
It's hard to follow what's going on here. What is this doing?

Reply:
Ooh I see! Instead of filtering twice, this can be simplified to 
```
return event.start == promotion.date &&
  event.channel == promotion.channel &&
  event.releaseName == promotion.releaseName
```

Unsure if those are the right property names but you get the gist!